File: HW/HW1/questionOne.java
package HW.HW1;

public class questionOne {
    public static boolean isPalindrome(String s) {

        String cleaned = s.replaceAll("[^A-Za-z0-9]", "").toLowerCase();
        String str="";
        for(int i=cleaned.length()-1;i>-1;i--){
            str+=cleaned.charAt(i);
        }

        return cleaned.equals(str);
    }

    public static void main(String[] args) {
        String str = "A man, a plan, a canal: Panama";
        System.out.println(isPalindrome(str));  // true
        
        String s2 = "race a car";
        System.out.println(isPalindrome(s2)); // false
    }
    
}

File: HW/HW1/questionOne.py
def is_palindrome(s):
    cleaned = ''.join(char.lower() for char in s if char.isalnum())

    return cleaned == cleaned[::-1]

if __name__ == "__main__":
    s = "A man, a plan, a canal: Panama"
    print(is_palindrome(s))
    
    s2 = "race a car"
    print(is_palindrome(s2))